### PR TITLE
Make Installer Genericer

### PR DESCRIPTION
**Pull request by papeh**
- Parameterize a few registry keys
- Consolidate patch/(base|update)/AppNoUi.wxs
- Normalize some whitespace (and other cleanup)

---
This change is [Reviewable](https://reviewable.io/reviews/sillsdev/genericinstaller/4)

### CODE REVIEW

File: BaseInstallerBuild/TemplateFramework.wxs [hunk 1]
```diff
@@ -1,54 +1,54 @@
 <?xml version="1.0"?>
 
-<!-- This file is used for creating a BASE product installation.  
-	 In other words this is for Paratext 8.0 only.  
+<!-- This file is used for creating a BASE product installation.
+	 In other words this is for Paratext 8.0 only.
```
Comment (jasonleenaylor) [inline BaseInstallerBuild/TemplateFramework.wxs]: In the future, we could make this read like an example

File: Common/Overrides.wxi [hunk 2]
```diff
@@ -0,0 +1,2 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (jasonleenaylor) [inline Common/Overrides.wxi]: Add to gitignore?

**Comment by marksvc** [comment]
FYI    I'm not very knowledgeable about this. I examined it and don't have objections. 

---

Reviewed 7 of 7 files at r1.
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---

*[BaseInstallerBuild/TemplateFramework.wxs, line 94 at r1](https://reviewable.io:443/reviews/sillsdev/genericinstaller/4#-KzUUXRY8GyBgsNJArv0:-KzUUXRZtWodVea33vdR:b-j5vnx4) ([raw file](https://github.com/sillsdev/genericinstaller/blob/2e9ef622cdcae5debe1033920316ebaff54cb755/BaseInstallerBuild/TemplateFramework.wxs#L94)):*
> not REGISTRYKEY

That seems to be an odd thing to write here. But I'm not familiar with the syntax. (And 3 other lines elsewhere in the changeset.)

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/sillsdev/genericinstaller/4#-:-KzUYQpuxSmk3x9l40ND:b-r4fqgq)*

**Comment by marksvc** [comment]
FYI (I wrote FYI because Reviewable implied that syntax was the right way to begin my comment, but it reads weird to me :). You can ignore the "FYI" when reading my comment.)

---

Review status: all files reviewed at latest revision, 3 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/sillsdev/genericinstaller/4#-:-KzUYueIbvUhpoXuWBag:b-6kjg37)*

**Comment by papeh** [comment]
*[BaseInstallerBuild/TemplateFramework.wxs, line 129 at r1](https://reviewable.io:443/reviews/sillsdev/genericinstaller/4#-KzUeVKvbO_vjviicg8Q:-KzUeVKwKcgDIpBpuGND:b-wrrrr5) ([raw file](https://github.com/sillsdev/genericinstaller/blob/2e9ef622cdcae5debe1033920316ebaff54cb755/BaseInstallerBuild/TemplateFramework.wxs#L129)):*
> ```XML
> 	<WixVariable Id="WixUIDialogBmp" Value="..\resources\MsiBackground.bmp" />
> 
> 	<SetProperty Id="APPFOLDER" Value="[ProgramFilesFolder]$(var.SafeApplicationName)$(var.MajorVersion)\" Before="AppSearch">not APPFOLDER</SetProperty>
> ```

@marksvc, I thought the syntax was a bit odd, too, but yes, the inner text is the condition and the value is in an attribute.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/sillsdev/genericinstaller/4)*

**Pull request merged**